### PR TITLE
Implements __str__ method for PyMI objects

### PR DESCRIPTION
**Pull request by claudiubelu**
Currently, PyMI debugging using PyMI can be a bit difficult, when compared to the old WMI.

For example, the old WMI has an object string representation implementation, meaning that WMI objects can easily and usefully be printed, while PyMI doesn't:

    >>> print new_wmi.Msvm_VirtualSystemSettingData()[0]
    <wmi._Instance object at 0x0000000002E45CC0>

This patch adds a similar __str__ implementation to PyMI.

### CODE REVIEW

File: PyMI/src/wmi/__init__.py [hunk 1]
```diff
@@ -218,6 +218,24 @@ def get_class_name(self):
     def get_class(self):
         pass
 
+    def __str__(self):
+        try:
+            obj = self.get_wrapped_object()
+
+            obj_string = 'instance if %s\n{' % obj.get_class_name()
```
Comment (petrutlucian94) [inline PyMI/src/wmi/__init__.py]: Small typo here, I guess you want 'instance of'

File: PyMI/src/wmi/__init__.py [hunk 2]
```diff
@@ -218,6 +218,24 @@ def get_class_name(self):
...
+
+            # an MI object's length is the number of attributes the object it
+            # represents has.
+            for i in range(len(obj)):
+                key, value_type, value = obj.get_element(i)
+                obj_string += "\n        %s = %s;" % (key, value)
```
Comment (petrutlucian94) [inline PyMI/src/wmi/__init__.py]: You can simply use '\t'

File: PyMI/src/wmi/__init__.py [hunk 3]
```diff
@@ -218,6 +218,24 @@ def get_class_name(self):
...
 
+    def __str__(self):
+        try:
+            obj = self.get_wrapped_object()
+
+            obj_string = 'instance if %s\n{' % obj.get_class_name()
+
+            # an MI object's length is the number of attributes the object it
+            # represents has.
+            for i in range(len(obj)):
```
Comment (petrutlucian94) [inline PyMI/src/wmi/__init__.py]: So you're using the same format that the old WMI lib had. Should we do something similar for methods as well?

The also have the 'properties' and 'methods' dict fields. Those may be useful as well and you may just use that when building the string representation.

**Comment by claudiubelu** [comment]
FWIW, this is how it looks now:

    >>> vms = conn.Msvm_ComputerSystem()
    >>> vms[0]
    <pymi_object: \\WIN-SRVR2\root\virtualization\v2:Msvm_ComputerSystem.Name="WIN-SRVR2",CreationClassName="Msvm_ComputerSystem">
    >>> print(vms[0])
    instance of Msvm_ComputerSystem
    {
            Caption = "Hosting Computer System";
            Description = "Microsoft Hosting Computer System";
            ElementName = "WIN-SRVR2";
            InstanceID = "None";
            CommunicationStatus = None;
            DetailedStatus = None;
            HealthState = 5;
            InstallDate = None;
            Name = "WIN-SRVR2";
            OperatingStatus = None;
            OperationalStatus = (2,);
            PrimaryStatus = None;
            Status = "OK";
            StatusDescriptions = ('OK',);
            AvailableRequestedStates = None;
            EnabledDefault = 2;
            EnabledState = 2;
            OtherEnabledState = "None";
            RequestedState = 12;
            TimeOfLastStateChange = None;
            TransitioningToState = None;
            CreationClassName = "Msvm_ComputerSystem";
            IdentifyingDescriptions = None;
            NameFormat = "None";
            OtherIdentifyingInfo = None;
            PrimaryOwnerContact = "None";
            PrimaryOwnerName = "None";
            Roles = None;
            Dedicated = None;
            OtherDedicatedDescriptions = None;
            PowerManagementCapabilities = None;
            ResetCapability = 1;
            EnhancedSessionModeState = None;
            FailedOverReplicationType = None;
            LastApplicationConsistentReplicationTime = None;
            LastReplicationTime = None;
            LastReplicationType = None;
            LastSuccessfulBackupTime = None;
            NumberOfNumaNodes = 1;
            OnTimeInMilliseconds = None;
            ProcessID = None;
            ReplicationHealth = None;
            ReplicationMode = None;
            ReplicationState = None;
            TimeOfLastConfigurationChange = None;
    };
    >>> vms[0].RequestStateChange
    <function RequestStateChange (RequestedState, TimeoutPeriod)>

**Comment by mickare** [comment]
Works like a charm, why is this not accepted?

**Pull request merged**